README.md

```diff
@@ -1 +1,155 @@
-# signalfx-nodejs
+[![NPM version][npm-image]][npm-url] [![Build Status](https://travis-ci.org/sheremetat/signalfx-nodejs.svg?branch=master)](https://travis-ci.org/sheremetat/signalfx-nodejs)
+# NodeJS client library for SignalFx
+
+This is a programmatic interface in JavaScript(CommonJS) for SignalFx's metadata and
+ingest APIs. It is meant to provide a base for communicating with
+SignalFx APIs that can be easily leveraged by scripts and applications
+to interact with SignalFx, or report metric and event data to SignalFx.
+
+
+## Installation
+
+To install using npm:
+```sh
+$ npm install signalfx --save-dev
+```
+
+
+## Usage
+
+### API access token
+
+To use this library, you need a SignalFx API access token, which can be
+obtained from the SignalFx organization you want to report data into.
+
+### Create client
+
+There are two ways to create a client object: 
+
++ Using the default constructor `SignalFx`. This constructor uses Protobuf to send data to SignalFx, and falls back to JSON if Protobuf is not possible.
++ Using the JSON constructor `SignalFxJson`. This constructor forces JSON format to send data to SignalFx.
+
+```js
+var signalfx = require('signalfx');
+
+// Create default client
+var client = new signalFx.SignalFx('MY_SIGNALFX_TOKEN');
+// or create JSON client
+var clientJson = new signalFx.SignalFxJson('MY_SIGNALFX_TOKEN');
+```
+
+### Reporting data
+
+This example shows how to report metrics to SignalFx as cumulative counters, gauges, and counters.
+```js
+var signalfx = require('signalfx');
+
+var client = new signalFx.SignalFx('MY_SIGNALFX_TOKEN');
+
+client.send({
+           cumulative_counters:[
+             {metric: 'myfunc.calls_cumulative', value: 10},
+             ...
+           ],
+           gauges:[
+             {metric: 'myfunc.time', value: 532},
+             ...
+           ],
+           counters:[
+             {metric: 'myfunc.calls', value: 42},
+             ...
+           ]});
+```
+
+You can also include dimensions with your metrics, as follows:
+
+```js
+var signalfx = require('signalfx');
+
+var client = new signalFx.SignalFx('MY_SIGNALFX_TOKEN');
+
+client.send({
+          cumulative_counters:[
+            {'metric': 'myfunc.calls_cumulative', 'value': 10, 'dimensions': {'host': 'server1', 'host_ip': '1.2.3.4'}},
+            ...
+          ],
+          gauges:[
+            {'metric': 'myfunc.time', 'value': 532, 'dimensions': {'host': 'server1', 'host_ip': '1.2.3.4'}},
+            ...
+          ],
+          counters:[
+            {'metric': 'myfunc.calls', 'value': 42, 'dimensions': {'host': 'server1', 'host_ip': '1.2.3.4'}},
+            ...
+          ]});
+```
+
+### Sending events
+
+Events can be sent to SignalFx via the `sendEvent` function. The
+event type must be specified, and dimensions and extra event properties
```

Comment:
@cep21 @mdubbyap Can you help me with it? I've just grubbed this string from signalfx-python library. Thank you in advance.

In case of nodejs library implementation event type is just an any string